### PR TITLE
Add Go 1.21

### PR DESCRIPTION
**Pull request by DmitriyMV**
Just like #4970 but for Go 1.21. Depends on compiler-explorer/infra#1067

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Go 1.21 didn't install, happy to liaise on that issue.

File: etc/config/go.amazon.properties [hunk 1]
```diff
@@ -110,11 +110,14 @@ compiler.gl1190.goroot=/opt/compiler-explorer/golang-1.19/go
 compiler.gl1200.exe=/opt/compiler-explorer/golang-1.20/go/bin/go
 compiler.gl1200.semver=1.20
 compiler.gl1200.goroot=/opt/compiler-explorer/golang-1.20/go
+compiler.gl1210.exe=/opt/compiler-explorer/golang-1.21/go/bin/go
```
Comment (mattgodbolt) [inline etc/config/go.amazon.properties:113]: This path is now out of date: the changes in infra means these paths are all now: `/opt/compiler-explorer/golang-1.21.0`

**Review by mattgodbolt — CHANGES_REQUESTED** [review]
Thank you! Unfortunately the infra change to the version means the destination path is now `/opt/compiler-explorer/golang-1.21.0` so these need updating. Hopefully a quick fix!

File: etc/config/go.amazon.properties (hunk 1 above)
Comment (DmitriyMV) [reply etc/config/go.amazon.properties]: Done!

**Review by mattgodbolt — APPROVED** [review]
Thanks!

**Pull request merged**